tests/test_bt_merge.cpp

```diff
@@ -24,14 +143,97 @@ TEST_CASE("bt_dict merging", "[bt_dict][merge]") {
     CHECK(session::bt::merge(bt_dict{}, bt_dict{}) == bt_dict{});
 }
 
+std::list<int64_t> unique_sorted(const std::list<int64_t>& list) {
+    std::set<int64_t> unique_values(list.begin(), list.end());
+    std::list<int64_t> sorted_list(unique_values.begin(), unique_values.end());
+    sorted_list.sort();
+    return sorted_list;
+}
+
+bt_list map_to_bt_list(const std::list<int64_t>& lst) {
+    bt_list result;
+    std::transform(lst.begin(), lst.end(), std::back_inserter(result), [](int64_t val) {
+        return bt_value{val};
+    });
+    return result;
+}
+
+bt_list gen_unique_sorted_bt_list() {
+    return map_to_bt_list(unique_sorted(*rc::gen::arbitrary<std::list<int64_t>>()));
+}
+
+// TODO: implement a generic version of compare()
+auto compare = [](const auto& a, const auto& b) {
```

Comment:
in C++20 there is `operator <=>`

Reply:
some submodule failed to build with C++20, I'll keep the old way for now. A generic version of compare and a generic version of bt_list merge_sorted() test is implemented.